google/cloud/pubsublite/cloudpubsub/internal/publisher_impl.py

```diff
@@ -42,7 +43,11 @@ def publish(
 
     def __enter__(self):
         self._managed_loop.__enter__()
-        self._managed_loop.submit(self._underlying.__aenter__()).result()
+        try:
+            self._managed_loop.submit(self._underlying.__aenter__()).result()
+        except GoogleAPICallError:
+            self._managed_loop.__exit__(None, None, None)
```

Comment:
You can't actually do this. This will result in the managed loop being closed twice when a context manager is used. You shouldn't try to catch this exception- the caller still has to call __exit__

Reply:
You're right, the problem is that we are not calling exit on SinglePublisherImpl from the MultiplexedPublisherClient.

The MultiplexedPublisherClient attempts to get_or_create a SinglePublisherImpl here: https://github.com/googleapis/python-pubsublite/blob/aa52ef4f4aaf1136c054750725c1cf3e6cdefff5/google/cloud/pubsublite/cloudpubsub/internal/multiplexed_publisher_client.py#L53

The ClientMultiplexer will attempt to create and start the client here:
https://github.com/googleapis/python-pubsublite/blob/aa52ef4f4aaf1136c054750725c1cf3e6cdefff5/google/cloud/pubsublite/cloudpubsub/internal/client_multiplexer.py#L43

But, in the Not Found topic case, the client fails on __enter__ because the call to getTopicPartitions fails.  So, this raises an exception and the new publisher client is never assigned to live_clients[key].

When exit is eventually called on the ClientMultiplexer, we will never call __exit__ on the 'not found' client because it was never assigned to live_clients[key]: https://github.com/googleapis/python-pubsublite/blob/aa52ef4f4aaf1136c054750725c1cf3e6cdefff5/google/cloud/pubsublite/cloudpubsub/internal/client_multiplexer.py#L74

Therefore, we will never run __exit__ on the managed_loop.  I will re-work this PR to address the real issue.